### ISSUE TITLE
ASVS v4

### ISSUE THREAD

**Issue by sydseter** [OWASP/cornucopia#13]
Any interest out there for creating a version of the game for ASVS 4.0?
We are using the game internally to be able to help in defining security constraints for our epics and user-stories ([ASVS v4, 1.1.3](https://github.com/OWASP/ASVS/blob/master/4.0/en/0x10-V1-Architecture.md#v11-secure-software-development-lifecycle)), but I find that I end up translating the ASVS v3 requirements into v4. I am realizing that creating additional cards for a ASVS 4.0 version would be the best.

**Comment by rewtd** [comment]
Hi @sydseter absolutely! Or creating a new ecommerce-mappings-2.0.yaml file with the mappings corrected on it? Once I get the test framework working again I'll jump into this.

**Comment by sydseter** [comment]
Great, you might want to consider exchanging some scenarios for new cards and perhaps even renaming or changing categories as well, but I can start with translating the easy parts first. Perhaps the rest can be discussed?

**Comment by sydseter** [comment]
[OWASP ASVS 3.0 & 4.0 Comparison mapped to Cornucopia.xlsx](https://github.com/OWASP/cornucopia/files/13887412/OWASP.ASVS.3.0.4.0.Comparison.mapped.to.Cornucopia.xlsx)

Hi, I have created a mapping for ASVS 3.0.1 to ASVS 4.0 that can be used with Cornucopia.

I have also finished implementing changes to the cli application so that it is possible to convert between different mappings versions so that it is possible to go back and publish updates to Cornucopia 1.21 ecommerce version if needed. I am doing the final testing and will create a pull-request momentarily. For now, please, if someone can look at the ASVS 3.0.1 to 4.0 mapping and comment where needed, then I can make some last minute adjustments. 

Should I add the idml, docx and pdf files to the output folder so that they can be reviewed?

I found several copy&past errors in the translations. I have fixed these as well.

**Comment by sydseter** [comment]
I could also give a hand in updating https://copi.securedelivery.io/ as well if you trust me and need a hand. We use it on a regular basis in our company.
I have had some issues getting others to facilitate the copi sessions. 
Updating the ASVS mapping to 4.0 would be a big help.

**Comment by sydseter** [comment]
I realized now that there is an error in the excel here, but it didn't spill over in ecommerce-mappings-1.3.yaml so I guess it's fine. 

Just want to point it out in case you decide to have a second look.

The error is found in the Excel under B7. The right value is:

1.6.4 Verify that the architecture treats client-side secrets--such as symmetric keys, passwords, or API tokens--as insecure and never uses them to protect or access sensitive data.

1.10 in ASVS 3.0 was split up and moved around in ASVS 4.0 so there is no clear equivalent, but I felt some requirement concerning the use of client-side secrets could be good to add to Data Validation and Encoding in addition, maybe that card should have ASVS4; 4.1.1 as well? (e.g: Verify that the application enforces access control rules on a trusted service layer, especially if client-side access control is present and could be bypassed.)

1.6.4 and 4.1.1 are the only two requirement in ASVS 4.0 that could cover 1.10, but I know I am stretching things a bit and in addition, 4.1.1 maps better to 2.4.

In short, I tried to use common sense where there were to clear candidate, but I  guess this is fine since it's a requirement game that is meant to spawn discussions after all.

**Comment by sydseter** [comment]
I see that the there are references to ASVS 3.0 in the documents that should be updated. I'll have a look at it.
